### PR TITLE
Add Unit Test infrastucture

### LINKED ISSUE #72: Add unittest infrastructure
We currently have no unittests, and there's a variety of places (e.g. around clock start/stop) that we would benefit from them. It seems that Junit works with Ant, so getting that working would be a good place to start: https://ant.apache.org/manual/Tasks/junit.html

### PR DESCRIPTION
**Pull request by official-sounding**
Add ability to run jUnit 4.x tests to both the ant file and the eclipse project.  Also add a simple example unit test that ensures that crg objects can be accessed.

Test compilation goes to it's own top-level bin directory so that the test classes don't get included in the main release, and I didn't set the test task as on the required path for an ant build or release.  Once we have meaningful unit tests, we can decide if passing tests are a requirement for PR merging/releases/etc.

Closes #72

### CODE REVIEW

File: .settings/org.eclipse.jdt.core.prefs [hunk 1]
```diff
@@ -1,7 +1,12 @@
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
+org.eclipse.jdt.core.compiler.codegen.methodParameters=do not generate
```
Comment (brian-brazil) [inline .settings/org.eclipse.jdt.core.prefs]: Are these changes related to this PR?

File: .project [hunk 2]
```diff
@@ -26,4 +26,15 @@
 		<nature>org.eclipse.jdt.core.javanature</nature>
 		<nature>org.eclipse.wst.jsdt.core.jsNature</nature>
 	</natures>
+	<filteredResources>
```
Comment (brian-brazil) [inline .project:29]: Are these changes related to this PR?

**Review by brian-brazil — COMMENTED** [review]
This is great! I'd make tests required for at least release, so we don't forget to do it later.

File: .project (hunk 2 above)
Comment (official-sounding) [reply .project:29]: Yes, this filters the test-bin directory from view in eclipse

**Comment by official-sounding** [comment]
removed the changes to the file in .settings with a force-push.  I think they were added inadvertently while debugging on my other PR.

File: build.xml [hunk 3]
```diff
@@ -37,6 +39,13 @@
...
 
+	
+	 <path id="classpath.test">
+	    <pathelement location="test-libs/junit-4.11.jar"/>
+	    <pathelement location="test-libs/hamcrest-core-1.3.jar"/>
+	    <pathelement location="${dest.dir}"/>
+	  </path>
+	
 	<!-- Show help (the different options for this build.xml) -->
 	<target name="help">
```
Comment (brian-brazil) [inline build.xml]: This needs updating too

**Review by brian-brazil — COMMENTED** [review]
Can you make this a dependency of the release and we can get this in?

**Comment by official-sounding** [comment]
cleaned up the tabs/spaces issues I introduced, made the test task a pre-req of the "zip" step that is used to make releases

**Comment by brian-brazil** [comment]
Thanks!
